src/main/java/io/sqreen/sasdk/backend/IngestionHttpClient.java

```diff
@@ -144,69 +145,96 @@
         void reportTrace(Object trace) throws IOException;
     }
 
-    static class IngestionHttpClientImpl
-            implements WithAuthentication, WithoutAuthentication {
 
-        private final String host;
-        private final BackendHttpImpl backendHttp;
-        private final AuthenticationConfig config;
+    static class IngestionHttpAuthClientImpl extends IngestionHttpClientImpl  implements WithAuthentication {
 
-        public IngestionHttpClientImpl(String host,
-                                       BackendHttpImpl backendHttp,
-                                       AuthenticationConfig config) {
-            checkNotNull(host);
-            checkNotNull(backendHttp);
-            this.host = host;
-            this.backendHttp = backendHttp;
+        protected final AuthenticationConfig config;
+        private Multimap<String, String> headers = ArrayListMultimap.create();
```

Comment:
Should be final and immutable so that it can be safely shared between threads (though in practice because there's one `final` field, I think current implementations will safely publish)